src/main/java/pantrypal/general/control/Parser.java

```diff
@@ -0,0 +1,112 @@
+package pantrypal.general.control;
+
+import pantrypal.general.commands.Command;
+import pantrypal.general.commands.HelpCommand;
+import pantrypal.general.commands.ExitCommand;
+import pantrypal.general.commands.NullCommand;
+import pantrypal.general.commands.inventory.AddIngredientCommand;
+import pantrypal.general.commands.inventory.IncreaseQuantityCommand;
+import pantrypal.general.commands.inventory.DecreaseQuantityCommand;
+import pantrypal.general.commands.inventory.SetAlertCommand;
+import pantrypal.general.commands.inventory.CheckStockCommand;
+import pantrypal.general.commands.inventory.ViewLowStockCommand;
+import pantrypal.general.commands.inventory.DeleteIngredientCommand;
+import pantrypal.general.commands.mealplan.AddNewPlan;
+import pantrypal.general.commands.mealplan.RemovePlan;
+import pantrypal.general.commands.mealplan.RemoveRecipeFromPlan;
+import pantrypal.general.commands.mealplan.ViewPlan;
+import pantrypal.general.commands.recipe.AddRecipe;
+import pantrypal.general.commands.recipe.ListRecipe;
+import pantrypal.general.commands.recipe.RemoveRecipe;
+import pantrypal.general.commands.recipe.ViewRecipe;
+import pantrypal.general.commands.shoppinglist.AddShoppingItem;
+import pantrypal.general.commands.shoppinglist.GenerateShoppingList;
+import pantrypal.general.commands.shoppinglist.RemoveShoppingItem;
+import pantrypal.general.commands.shoppinglist.ViewShoppingList;
+import pantrypal.general.commands.mealplan.AddRecipeToPlan;
+
+public class Parser {
+
+    public Command parse(String input) {
```

Comment:
function is >30LOC so try to break it up into multiple functions/refactor similar portions of code into new methods.